webapp/src/components/channel_header_actions/channel_header_menu.jsx

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import JiraIcon from 'components/icon';
+
+export default class ChannelHeaderMenuAction extends React.PureComponent {
+    static propTypes = {
+        open: PropTypes.func.isRequired,
+        channelId: PropTypes.string.isRequired,
+        isInstanceInstalled: PropTypes.bool.isRequired,
+        userConnected: PropTypes.bool.isRequired,
+    };
```

Comment:
Mind making this file typescript? Don't worry about making `index.js` typescript if it gives you trouble. I have picking up Allan's PR to typescript migration on my backlog.